packages/canon/src/components/DataTable/Pagination/DataTablePagination.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { forwardRef } from 'react';
+import { Text } from '../../Text';
+import { DataTablePaginationProps } from './types';
+import { IconButton } from '../../IconButton';
+import clsx from 'clsx';
+import { Select } from '../../Select';
+
+/** @public */
+function DataTablePagination<TData>(
+  props: DataTablePaginationProps<TData>,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { className, table, ...rest } = props;
+  const pageIndex = table?.getState().pagination.pageIndex;
+  const pageSize = table?.getState().pagination.pageSize;
+
+  return (
+    <div
+      ref={ref}
+      className={clsx('canon-TablePagination', className)}
+      {...rest}
+    >
+      <div className="canon-TablePagination--left">
+        <Select
+          name="pageSize"
+          size="small"
+          placeholder="Show 10 results"
+          options={[
```

Comment:
ofc important for the caller to be able to set the page size too, as it can often be limited by the backing api

Reply:
You mean adding a `pageSize` prop to the `DataTablePagination` component' and use it as a default? Or you mean that this will force the `pageSize` and disable update buttons?

Reply:
All data + state can be controlled from the `useReactTable` hook directly. I added a story for a controlled experience.